### PR TITLE
CBG-605 Read throughput optimizations

### PR DESCRIPTION
**Pull request by adamcfraser**
Collection of read throughput enhancements, based on toy build testing.  Per-commit results available in Mercury perf testing spreadsheet.

Additional details included with each commit below (shouldn't squash on merge).

- [ ] http://uberjenkins.sc.couchbase.com:8080/view/Build/job/sync-gateway-integration/33/

### CODE REVIEW

**Comment by adamcfraser** [comment]
Fixes, rebased on master, re-running integration tests.

**Review by bbrks — COMMENTED** [review]
Couple of minor comments for a test and constant usage.

File: db/changes.go [hunk 1]
```diff
@@ -132,11 +132,11 @@ func (db *Database) addDocToChangeEntry(entry *ChangeEntry, options ChangesOptio
 }
 
 func (db *Database) AddDocToChangeEntryUsingRevCache(entry *ChangeEntry, revID string) (err error) {
-	rev, err := db.getRev(entry.ID, revID, 0, nil)
+	rev, err := db.getRev(entry.ID, revID, 0, nil, RevCacheIncludeBody)
 	if err != nil {
 		return err
 	}
-	entry.Doc, err = rev.As1xBytes(db, nil, nil, false)
+	entry.Doc, err = rev.As1xBytes(db, nil, nil, RevCacheOmitBody)
```
[the remark this answers is not in the document]
Comment (adamcfraser) [reply db/changes.go]: Agreed - will fix

File: db/crud_test.go [hunk 2]
```diff
@@ -49,6 +49,42 @@ func getRevTreeList(bucket base.Bucket, key string, useXattrs bool) (revTreeList
...
+	assert.NoError(t, err, "Couldn't get document")
+
+	docRev, err := db.GetRev("doc1", "1-a", false, nil)
+
+	bodyBytes, err := base.InjectJSONProperties(docRev.BodyBytes, base.KVPair{Key: "test", Val: "abc"})
+	log.Printf("First bytes: %s", bodyBytes)
+
+	docRevAgain, err := db.GetRev("doc1", "1-a", false, nil)
+	secondBytes, err := base.InjectJSONProperties(docRevAgain.BodyBytes, base.KVPair{Key: "test2", Val: "abc"})
+	log.Printf("Second bytes: %s", secondBytes)
```
[the remark this answers is not in the document]
Comment (adamcfraser) [reply db/crud_test.go]: This test was primarily to facilitate stepping through the code to see where marshalling would happen.  Let me review to see whether it merits additional assertions.

**Review by bbrks — APPROVED** [review]

**Pull request merged**